examples/erc1155/src/lib.rs

```diff
@@ -42,4 +46,9 @@ impl Erc1155Example {
         self.erc1155._mint_batch(to, token_ids, amounts, &data)?;
         Ok(())
     }
+
+    #[selector(name = "setURI")]
+    pub fn set_uri(&mut self, new_uri: String) {
```

Comment:
`setURI` is neither part of ERC-1155 nor Metadata URI extension. You should set URI only in the constructor for this extension.